modules/serverless-kafka-broker-sasl-default-config.adoc

```diff
@@ -0,0 +1,98 @@
+// Module is included in the following assemblies:
+//
+// * serverless/knative_eventing/serverless-kafka.adoc
+
+[id="serverless-kafka-broker-sasl-default-config_{context}"]
+= Configuring SASL authentication for the Kafka broker with default configuration
+
+.Prerequisites
+
+* The {ServerlessOperatorName}, Knative Eventing, and the `KnativeKafka` custom resource are installed on your {product-title} cluster.
+* You have created a project or have access to a project with the appropriate roles and permissions to create applications and other workloads in {product-title}.
+* A username and password for the Kafka cluster.
+* Choose the SASL mechanism to use, for example `PLAIN`, `SCRAM-SHA-256`, or `SCRAM-SHA-512`.
+* If TLS is enabled, you also need the `ca.crt` certificate file for the Kafka cluster.
+
+[NOTE]
+====
+Red Hat recommends to enable TLS in addition to SASL.
+====
+
+.Procedure
+
+. Create the certificate files as secrets in the `knative-eventing` namespace:
++
+[source,terminal]
+----
+$ oc create secret --namespace knative-eventing generic <kafka_auth_secret> \
+  --from-literal=protocol=SASL_SSL \
+  --from-literal=sasl.mechanism=<sasl_mechanism> \
+  --from-file=ca.crt=caroot.pem \
+  --from-literal=password="SecretPassword" \
+  --from-literal=user="my-sasl-user"
+----
++
+[IMPORTANT]
+====
+Use the key names `ca.crt`, `password`, and `sasl.mechanism`. Do not change them.
+====
+
+. Start editing the `KnativeKafka` custom resource:
++
+[source,terminal]
+----
+$ oc edit knativekafka
+----
+
+. Reference your secret and the namespace of the secret:
++
+[source,yaml]
+----
+apiVersion: operator.serverless.openshift.io/v1alpha1
+kind: KnativeKafka
+metadata:
+  namespace: knative-eventing
+  name: knative-kafka
+spec:
+  broker:
+    enabled: true
+    defaultConfig:
+      bootstrapServers: <bootstrap_servers>
+      numPartitions: <num_partitions>
+      replicationFactor: <replication_factor>
+      authSecretName: <kafka_auth_secret>
+  channel:
+    enabled: false
+  source:
+    enabled: false
```

Comment:
Call out the secret and namespace in this example.

Reply:
@lpettyjo with "call out", do you mean adding numbers to them, and providing some description below the code snippet, like here: https://github.com/openshift/openshift-docs/pull/39408/files#diff-e37a5088c4981ab85e59a970a409fc334c60bf2a047d71c8065dbc14a832da03R52-R55  ?

Reply:
I'm not sure that this is necessarily required - we don't use callouts in a lot of other sample YAML too, and I think it's OK. I would recommend though spelling out what the variables are, e.g. rather than `<num_partitions>` maybe use `<number_of_partitions>` if it doesn't change the meaning, instead of `<kafka_auth_secret>` use `<kafka_authentication_secret>`, etc. Using better described variables in a lot of our other content is maybe the reason callouts aren't always required.

Reply:
@lpettyjo can you clarify whether the callouts Matthias is describing is what you mean? I've been assuming that's the case but maybe you meant something else? Otherwise yes @matzew , the only callouts I'm aware of are those.